avoid buffer overrun in UConverter_fromUnicode_ISO_2022_CN_OFFSETS_LOGIC
http://bugs.icu-project.org/trac/ticket/12389
--- icu4c-77_1/icu/source/common/ucnv2022.cpp
+++ icu4c-77_1/icu/source/common/ucnv2022.cpp
@@ -3107,8 +3107,11 @@ len = 0;
                                 }
                             }
                         } else {
+			    if (cs0>=UCNV_2022_MAX_CONVERTERS) {
+				*err =U_BUFFER_OVERFLOW_ERROR;
+				break;
+			    }
                             /* GB2312_1 or ISO-IR-165 */
-                            U_ASSERT(cs0<UCNV_2022_MAX_CONVERTERS);
                             len2 = MBCS_FROM_UCHAR32_ISO2022(
                                         converterData->myConverterArray[cs0],
                                         sourceChar,
@@ -3392,7 +3395,10 @@ tempBuf[2] = static_cast<char>(trailByte
                                 tempBufLen = 3;
 
                             }else{
-                                U_ASSERT(tempState<UCNV_2022_MAX_CONVERTERS);
+				if (tempState>=UCNV_2022_MAX_CONVERTERS) {
+					*err =U_BUFFER_OVERFLOW_ERROR;
+					break;
+				}
                                 cnv = myData->myConverterArray[tempState];
                                 tempBuf[0] = static_cast<char>(mySourceChar);
                                 tempBuf[1] = static_cast<char>(trailByte);
